src/target/riscv/riscv-013.c

```diff
@@ -1436,6 +1436,30 @@ static int execute_fence(struct target *target)
 	return result;
 }
 
+static int execute_fence_i(struct target *target)
```

Comment:
Do you still need this?

I think you can just replace all calls to execute_fence() with execute_fences().